### PR TITLE
Fix AppSet topology displaying resources as not deployed

### PR DESCRIPTION
**Pull request by fxiang1**
Signed-off-by: Feng Xiang <fxiang@redhat.com>

https://issues.redhat.com/browse/ACM-3159

- Fix combining arrays resulting in an empty array

### CODE REVIEW

**Review by chenz4027 — APPROVED** [review]
/lgtm

**Comment by fxiang1** [comment]
Waiting for approval for 2.6.4. FYI @cbynum-redhat .

**Pull request merged**
